SF iOS/SF iOSTests/EventDataSourceTests.m

```diff
@@ -7,40 +7,56 @@
 //
 
 #import <XCTest/XCTest.h>
-#import "EventDataSource.h"
-
-@interface EventDataSourceTests : XCTestCase
+#import "EventDataSource+RealmConfiguration.h"
+#import <Realm/Realm.h>
 
+@interface EventDataSourceTests : XCTestCase <EventDataSourceDelegate>
+@property (nonatomic) EventDataSource *dataSource;
+@property (nonatomic) XCTestExpectation *fetchingExpectation;
 @end
 
 @implementation EventDataSourceTests
 
 - (void)setUp {
+    _fetchingExpectation = [self expectationWithDescription:@"Wait for events"];
+
+    RLMRealmConfiguration *realmConfiguration = [RLMRealmConfiguration defaultConfiguration];
+    
+    // In-memory Realms do not save data across app launches, but all other features of Realm will work as expected, including querying, relationships and thread-safety. This is a useful option if you need flexible data access without the overhead of disk persistence.
+    [realmConfiguration setInMemoryIdentifier:@"UnitTest"];
+
+    [RLMRealmConfiguration setDefaultConfiguration:realmConfiguration];
+    
+    self.dataSource = [[EventDataSource alloc] initWithEventType:EventTypeSFCoffee withRealmConfiguration:realmConfiguration];
+    
+    self.dataSource.delegate = self;
+    
     [super setUp];
 }
 
+- (void)testFetchingCoffeeEvents {
+    XCTAssertEqual(self.dataSource.numberOfEvents, 0);
+    [self.dataSource refresh];
```

Comment:
Here in `refresh` we're fetching data from the network (see the title of the test case) which makes the tests nondeterministic and based on some global state that we can't control (internet connectivity). What you want to do here is just add a few items to the realm, then validate the delegate callbacks. The datasource is designed to skip the network altogether. This'll take a bit of work but I think we can definitely be more deterministic. I'll take a quick stab to get you going in the right direction, standby.